presto-memory/src/test/java/io/prestosql/plugin/memory/TestMemorySmoke.java

```diff
@@ -161,6 +161,20 @@ public void testJoinDynamicFilteringSingleValue()
                 ImmutableSet.of(39, PART_COUNT));
     }
 
+    @Test
+    public void testJoinDynamicFilteringImplicitCoercion()
+    {
+        assertUpdate("CREATE TABLE coerce_test AS SELECT CAST(orderkey as INT) orderkey_int FROM tpch.tiny.lineitem", "SELECT count(*) FROM lineitem");
+        assertQueryResult("SELECT orderkey FROM orders WHERE comment = 'nstructions sleep furiously among '", 1L);
```

Comment:
no need to revalidate these. These are validated as part of other tests (e.g `testJoinDynamicFilteringSingleValue`)